qa/integration-tests/src/test/java/io/camunda/it/client/IncidentQueryTest.java

```diff
@@ -0,0 +1,514 @@
+  static void beforeAll() throws InterruptedException {
+    zeebeClient = testStandaloneCamunda.newClientBuilder().build();
+
+    final var processes =
+        List.of("service_tasks_v1.bpmn", "service_tasks_v2.bpmn", "incident_process_v1.bpmn");
+    processes.forEach(
+        process ->
+            DEPLOYED_PROCESSES.addAll(
+                deployResource(String.format("process/%s", process)).getProcesses()));
+
+    waitForProcessesToBeDeployed();
+
+    PROCESS_INSTANCES.add(startProcessInstance("service_tasks_v1"));
+    PROCESS_INSTANCES.add(startProcessInstance("service_tasks_v2", "{\"path\":222}"));
+    PROCESS_INSTANCES.add(startProcessInstance("incident_process_v1"));
+
+    waitForProcessInstancesToStart(3);
+    waitForProcessInstancesToExecute(3);
+
+    incident = zeebeClient.newIncidentQuery().send().join().items().getFirst();
+  }
+
+  @Test
+  void shouldGetIncidentByKey() {
+    // given
+    final var incidentKey = incident.getIncidentKey();
+    // when
+    final var result = zeebeClient.newIncidentGetRequest(incidentKey).send().join();
+
+    // then
+    assertThat(result).isNotNull();
+    assertThat(result).isEqualTo(incident);
+  }
+
+  @Test
+  void shouldThrownExceptionIfNotFoundByKey() {
+    // given
+    final var invalidIncidentKey = 0xCAFE;
+
+    // when / then
+    final var exception =
+        assertThrowsExactly(
+            ProblemException.class,
+            () -> zeebeClient.newIncidentGetRequest(invalidIncidentKey).send().join());
+    assertThat(exception.getMessage()).startsWith("Failed with code 404");
+    assertThat(exception.details()).isNotNull();
+    assertThat(exception.details().getTitle()).isEqualTo("NOT_FOUND");
+    assertThat(exception.details().getStatus()).isEqualTo(404);
+    assertThat(exception.details().getDetail()).isEqualTo("Incident with key 51966 not found");
+  }
+
+  @Test
+  void shouldFilterByKey() {
+    // given
+    final var incidentKey = incident.getIncidentKey();
+
+    // when
+    final var result =
+        zeebeClient.newIncidentQuery().filter(f -> f.incidentKey(incidentKey)).send().join();
+
+    // then
+    assertThat(result.items().size()).isEqualTo(1);
+    assertThat(result.items().getFirst().getIncidentKey()).isEqualTo(incidentKey);
+  }
+
+  @Test
+  void shouldFilterByProcessInstanceKey() {
+    // given
+    final var processInstanceKey = incident.getProcessInstanceKey();
+
+    // when
+    final var result =
+        zeebeClient
+            .newIncidentQuery()
+            .filter(f -> f.processInstanceKey(processInstanceKey))
+            .send()
+            .join();
+
+    // then
+    assertThat(result.items().size()).isEqualTo(1);
+    assertThat(result.items().getFirst().getProcessInstanceKey()).isEqualTo(processInstanceKey);
+  }
+
+  // TODO:
+  // - processDefinitionKey
+  // - processDefinitionId
+  // - errorType
+  // - errorMessage
+  // - flowNodeId
+  // - creationTime
+  // - jobKey
+  // - treePath
+  // - tenantId
```

Comment:
🙃 Looks to me like you added those, but please double-check again

Reply:
Yes, came from GitHub Copilot. Need to clean this up. 🙈